hpx/components/containers/coarray/coarray.hpp

```diff
@@ -0,0 +1,240 @@
+//  Copyright (c) 2017 Antoine Tran Tan
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file hpx/components/coarray/coarray.hpp
+
+#ifndef HPX_COARRAY_HPP
+#define HPX_COARRAY_HPP
+
+#include <hpx/components/containers/coarray/detail/are_integral.hpp>
+#include <hpx/components/containers/coarray/detail/last_element.hpp>
+#include <hpx/components/containers/partitioned_vector/partitioned_vector.hpp>
+#include <hpx/components/containers/partitioned_vector/partitioned_vector_view.hpp>
+#include <hpx/lcos/spmd_block.hpp>
+#include <hpx/util/detail/pack.hpp>
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <initializer_list>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+///////////////////////////////////////////////////////////////////////////////
+/// \cond NOINTERNAL
+
+#define HPX_REGISTER_COARRAY(T) HPX_REGISTER_PARTITIONED_VECTOR(T)
```

Comment:
`HPX_REGISTER_PARTITIONED_VECTOR` is a variadic macro. Should `HPX_REGISTER_COARRAY` be variadic as well?